common/utils.go

```diff
@@ -78,6 +79,25 @@ type metadataWrapper struct {
 	Metadata Metadata `json:"compute"`
 }
 
+// HTTPClient interface to abstract http.Client methods
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// StandardHTTPClient is a standard implementation of the HTTPClient interface
+type StandardHTTPClient struct{}
+
+// Do is the implementation of the Post method for StandardHTTPClient
+func (c *StandardHTTPClient) Do(req *http.Request) (*http.Response, error) {
+	client := http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "http request failed")
+	}
+
+	return resp, nil
+}
+
```

Comment:
we don't need this - the same discussion was happening on [this PR](https://github.com/Azure/azure-container-networking/pull/2421#discussion_r1419463500)

Reply:
https://github.com/Azure/azure-container-networking/pull/2421#discussion_r1412577352

Reply:
Addressed.